src/nuis/measurement/Projection.h

```diff
@@ -0,0 +1,195 @@
+// Copyright 2016-2021 L. Pickering, P Stowell, R. Terri, C. Wilkinson, C. Wret
+
+/*******************************************************************************
+ *    This file is part of NUISANCE.
+ *
+ *    NUISANCE is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    NUISANCE is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with NUISANCE.  If not, see <http://www.gnu.org/licenses/>.
+ *******************************************************************************/
+#pragma once
+#include <vector>
+#include <string>
+#include <iostream>
+#include <memory>
+#include <map>
+
+#include <Eigen/Dense>
+
+#include "yaml-cpp/yaml.h"
+using namespace YAML;
+
+#include "nuis/measurement/Variables.h"
+#include "nuis/measurement/Document.h"
+
+enum likelihood_form_enum {
```

Comment:
I am quite very strongly of the opinion that we shouldn't have any stats stuff in here, the projections and bands should not know about stats at all, there should be other classes/functors/methods for calculating stats that should be allowed access to the data from these objects.